src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceProgress.cs

```diff
@@ -28,10 +29,19 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
             // destroy the data structures representing outstanding progress records
             // take down and destroy the progress display
 
+            if (_progPaneUpdateTimer != null)
+            {
+                // Stop update a progress pane and destroy timer
+                _progPaneUpdateTimer.Dispose();
+                _progPaneUpdateTimer = null;
+            }
+            progPaneUpdateFlag = false;
+
             if (_progPane != null)
             {
                 Dbg.Assert(_pendingProgress != null, "How can you have a progress pane and no backing data structure?");
 
+                // lock (_instanceLock) is not needed here because lock is done at global level
```

Comment:
There are two entry points:
1. `HandleIncomingProgressRecord` - here there are `lock (_instanceLock)`
2. ResetProgress - here there is no `lock (_instanceLock)`. This looks like a bug. I wanted to add the lock, but then discovered that there is a global lock in calling functions.